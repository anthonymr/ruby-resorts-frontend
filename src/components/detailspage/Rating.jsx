import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '@mui/material';
import {
  CircularProgressbarWithChildren,
  buildStyles,
} from 'react-circular-progressbar';
import RadialSeparators from './RadialSeparator';
import 'react-circular-progressbar/dist/styles.css';

const Rating = ({ rating }) => {
  let color = '#96bf01';
  if (rating < 3.5) {
    color = '#ffbb00';
  }
  if (rating < 2.5) {
    color = '#ff0000';
  }

  return (
    <Box
      sx={{
        width: { xs: '5rem', md: '3rem', lg: '5rem' },
        height: { xs: '5rem', md: '3rem', lg: '5rem' },
      }}
    >
      <CircularProgressbarWithChildren
        value={rating}
        text={`${rating}`}
        maxValue={5}
        strokeWidth={10}
        styles={buildStyles({
          strokeLinecap: 'butt',
          pathColor: color,
          textColor: color,
        })}
      >
        <RadialSeparators
          count={5}
          style={{
            background: '#fff',
            width: '3px',
            height: `${10}%`,
          }}
        />
      </CircularProgressbarWithChildren>
    </Box>
  );
};

Rating.propTypes = {
  rating: PropTypes.number,
};

Rating.defaultProps = {
  rating: 5,
};

export default Rating;
