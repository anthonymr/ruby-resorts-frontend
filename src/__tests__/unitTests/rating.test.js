import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Rating from '../../components/detailspage/Rating';

it('renders correctly', () => {
  const rating = renderer
    .create(
      <Router>
        <Rating />
      </Router>,
    )
    .toJSON();
  expect(rating).toMatchSnapshot();
});
