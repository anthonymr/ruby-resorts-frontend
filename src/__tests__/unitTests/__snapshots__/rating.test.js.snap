// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-167yxyd"
>
  <div
    data-test-id="CircularProgressbarWithChildren"
  >
    <div
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <svg
        className="CircularProgressbar "
        data-test-id="CircularProgressbar"
        style={Object {}}
        viewBox="0 0 100 100"
      >
        <path
          className="CircularProgressbar-trail"
          d="
      M 50,50
      m 0,-45
      a 45,45 0 1 1 0,90
      a 45,45 0 1 1 0,-90
    "
          fillOpacity={0}
          strokeWidth={10}
          style={
            Object {
              "strokeDasharray": "282.7433388230814px 282.7433388230814px",
              "strokeDashoffset": "0px",
              "strokeLinecap": "butt",
            }
          }
        />
        <path
          className="CircularProgressbar-path"
          d="
      M 50,50
      m 0,-45
      a 45,45 0 1 1 0,90
      a 45,45 0 1 1 0,-90
    "
          fillOpacity={0}
          strokeWidth={10}
          style={
            Object {
              "stroke": "#96bf01",
              "strokeDasharray": "282.7433388230814px 282.7433388230814px",
              "strokeDashoffset": "0px",
              "strokeLinecap": "butt",
            }
          }
        />
        <text
          className="CircularProgressbar-text"
          style={
            Object {
              "fill": "#96bf01",
            }
          }
          x={50}
          y={50}
        >
          5
        </text>
      </svg>
      <div
        data-test-id="CircularProgressbarWithChildren__children"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "center",
            "marginTop": "-100%",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "transform": "rotate(0turn)",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fff",
                "height": "10%",
                "width": "3px",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "transform": "rotate(0.2turn)",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fff",
                "height": "10%",
                "width": "3px",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "transform": "rotate(0.4turn)",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fff",
                "height": "10%",
                "width": "3px",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "transform": "rotate(0.6000000000000001turn)",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fff",
                "height": "10%",
                "width": "3px",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "transform": "rotate(0.8turn)",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fff",
                "height": "10%",
                "width": "3px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
